From f65caa1776dbc6b996f9f7af761b16ed1eef51cc Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Mon, 29 Jan 2018 10:01:39 +0800
Subject: [PATCH 3/4] Revert "Fix powerhint for NULL parameter"

 * This caused sending PROFILE_POWER_SAVE as a NULL pointer and
   crash the power HAL.

This reverts commit e0c1bbbe6d80bd7c3d266d36d18e87dd93bf98e1.

Change-Id: I8481f4a1d0e567cd39315e9cca2e26e7c8daa900
---
 power/1.0/default/Power.cpp | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/power/1.0/default/Power.cpp b/power/1.0/default/Power.cpp
index 51f87f5..2bc15fa 100644
--- a/power/1.0/default/Power.cpp
+++ b/power/1.0/default/Power.cpp
@@ -47,12 +47,8 @@ Return<void> Power::setInteractive(bool interactive)  {
 
 Return<void> Power::powerHint(PowerHint hint, int32_t data)  {
     int32_t param = data;
-    if (mModule->powerHint) {
-        if (data)
-            mModule->powerHint(mModule, static_cast<power_hint_t>(hint), &param);
-        else
-            mModule->powerHint(mModule, static_cast<power_hint_t>(hint), NULL);
-    }
+    if (mModule->powerHint)
+        mModule->powerHint(mModule, static_cast<power_hint_t>(hint), &param);
     return Void();
 }
 
-- 
2.7.4

